vms/proposervm/state_syncable_vm_test.go

```diff
@@ -590,3 +594,134 @@ func TestStateSummaryAcceptOlderBlock(t *testing.T) {
 	require.NoError(err)
 	require.Equal(proBlk.ID(), lastAcceptedID)
 }
+
+// TestStateSummaryAcceptOlderBlockSkipStateSync tests the case where we accept
+// a state summary older than the last accepted block. In this case, we should not
+// roll the ProposerVM back to match the state summary, but we should invoke the
+// innerVM to accept the state summary and re-align the ProposerVM with the innerVM
+// during the transition out of state sync.
+func TestStateSummaryAcceptOlderBlockSkipStateSync(t *testing.T) {
+	require := require.New(t)
+
+	innerVM, vm := helperBuildStateSyncTestObjects(t)
+	defer func() {
+		require.NoError(vm.Shutdown(context.Background()))
+	}()
+
+	// store post fork block associated with summary
+	innerBlk1 := &snowmantest.Block{
+		Decidable: snowtest.Decidable{
+			IDV: ids.GenerateTestID(),
+		},
+		BytesV:  []byte{1},
+		ParentV: ids.GenerateTestID(),
+		HeightV: 1969,
+	}
+	innerBlk2 := snowmantest.BuildChild(innerBlk1)
+
+	innerSummary1 := &blocktest.StateSummary{
+		IDV:     innerBlk1.ID(),
+		HeightV: innerBlk1.Height(),
+		BytesV:  innerBlk1.BytesV,
+	}
+
+	require.NoError(vm.SetForkHeight(innerSummary1.Height() - 1))
+
+	// Set the last accepted block height to be higher that the state summary
+	// we are going to attempt to accept
+	vm.lastAcceptedHeight = innerBlk2.Height()
+
+	innerVM.LastAcceptedF = func(context.Context) (ids.ID, error) {
+		return innerBlk2.IDV, nil
+	}
+
+	innerVM.GetBlockF = func(_ context.Context, blkID ids.ID) (snowman.Block, error) {
+		switch blkID {
+		case snowmantest.GenesisID:
+			return snowmantest.Genesis, nil
+		case innerBlk1.ID():
+			return innerBlk1, nil
+		case innerBlk2.ID():
+			return innerBlk2, nil
+		default:
+			return nil, database.ErrNotFound
+		}
+	}
+	innerVM.GetStateSummaryF = func(context.Context, uint64) (block.StateSummary, error) {
+		return innerSummary1, nil
+	}
+	innerVM.ParseBlockF = func(_ context.Context, b []byte) (snowman.Block, error) {
+		switch {
+		case bytes.Equal(b, innerBlk1.BytesV):
+			return innerBlk1, nil
+		case bytes.Equal(b, innerBlk2.BytesV):
+			return innerBlk2, nil
+		default:
+			panic("unexpected parse block")
+		}
+	}
+	// Setup innerSummary.AcceptF to return StateSyncSkipped and check it was called
+	calledInnerAccept := false
+	innerSummary1.AcceptF = func(context.Context) (block.StateSyncMode, error) {
+		calledInnerAccept = true
+		return block.StateSyncSkipped, nil
+	}
+
+	slb1, err := statelessblock.Build(
```

Comment:
Can we just call this `blk` or `statelessBlk` of `slb`? I don't think we've used this naming before

Reply:
We use this naming convention throughout the rest of the file. The only difference here is that we need two blocks and use 1/2 suffixes.